specs/speculative_specification.md

```diff
@@ -0,0 +1,67 @@
+The purpose of this document is to lay out the high-level view of pyiron's purposes, capabilities, and structure.
+Some what's written here represents our ambitions for 2023 and does not necessarily represent the current state of the code base; these ambitional aspects should be clearly indicated as such.
+
+# Why pyiron -- philosophy for prospective users
+
+pyiron exists to facilitate and accelerate computational research.
+
+In particular, pyiron...
+- makes easy things easy, and novel things possible
+- provides reliability and reproducibility by coupling results to the input and code(TODO) that generated them 
+- gives users tools to rapidly prototype new ideas, and easily scale those ideas up
+- abstracts away all (or at least most) of the nitty-gritty to let scientists focus on the science
+
+
+# What pyiron -- promises for actual users
+
+Pyiron provides a python framework for designing and managing research calculations.
+It is structured into a network of repositories: `pyiron_base`, which provides the core pyiron infrastructure, and then a variety of `pyiron_X` modules which build all the domain-specific tools for particular fields of research.
+E.g. `pyiron_atomistics` support for atomistic-scale materials science calculations, allowing users to create atomistic structures and perform a variety of calculations (minimization, molecular dynamics (MD), and more sophisticated routines such as nudged elastic band, variance-constrained semi-grand-canonical MD, etc.) using a variety of different engines to calculate energies, forces, and perhaps electronic state (e.g. LAMMPS, VASP, SPHiNX, QuantumEspresso...).
+Most of these extensions currently focus on materials science, but this is purely historical and `pyiron_base` is designed to support.
+
+Pyiron has a single point of entry, a single import, the `Project` object, from which all other pyiron objects can be created.
+Entire projects can be packaged and shared with colleagues.
+`Project` can either be imported from `pyiron_base` or from one of the other domain specific modules.
+When importing additional `pyiron_X` modules, the `Project` object is _automatically powered-up_ and given access to all the tools from that module.
+By heavy use of tab-completion, users can explore what tools are available from their project without having to immediately resort to reading documentation. 
```

Comment:
If we are talking about tab-completion, I would like to extend it to the triple pillar that I mention sometimes:

- Tab-completion
- Doc string
- Do and fail properly

In words: User checks all available options via tab-completion, chooses one of them and takes a look at the doc string to see what are the next steps. And they can execute the function according to what they think is right and the error message will guide them how to use it correctly, if it happened to not be the correct way to use it.

Reply:
From today's meeting, I would like add [a global picture](https://scikit-learn.org/stable/tutorial/machine_learning_map/index.html) that @liamhuber mentioned.

Reply:
Not adding much to this, but we could also consider [this figure displaying the LAMMPS class topology](https://docs.lammps.org/Developer_org.html#class-topology). At least for me, this is kind of easier to understand as im much more familiar with LAMMPS than with scikit-learn estimators.